src/hooks/useFetch.ts

```diff
@@ -1,58 +1,56 @@
 import { Dispatch, SetStateAction, useEffect, useState } from "react";
 
-interface FetchState<T>{
-    data: T | null,
-    loading: boolean,
-    error: string | null   ,
-    setData : Dispatch<SetStateAction<T | null>>,
-    setIsChanged : Dispatch<SetStateAction<boolean>>,
+interface FetchState<T> {
+  data: T | null;
+  loading: boolean;
+  error: string | null;
+  setData: Dispatch<SetStateAction<T | null>>;
+  setIsChanged: Dispatch<SetStateAction<boolean>>;
+  trigger: () => void;
 }
 
-const useFetch = <T>(fetchFun: ()  => Promise<T>):FetchState<T> => {
-    const [data,setData] = useState<T | null>(null)
-    const [loading,setLoading] = useState<boolean>(true)
-    const [isChanged,setIsChanged] = useState<boolean>(true)
-    const [error,setError] =useState<string|null>(null)
+const useFetch = <T>(fetchFun: () => Promise<T>): FetchState<T> => {
```

Comment:
This is good for learning and practicing, but for real-life applications. It is more common to use packages like react-query, SWR, redux-toolkit-query. These packages handle more than just fetching the data  and handling loading and error state. It adds functionalities like caching the data, preftetching, refetch on window focus, internet reconnection and retry on error, and more.  I recommend starting learning react-query as it will make the code more robust and it handles asynchronous operations like api calls efficiently